src/Missings.jl

```diff
@@ -514,4 +514,77 @@ julia> emptymissing(first)([1], 2)
 """
 emptymissing(f) = (x, args...; kwargs...) -> isempty(x) ? missing : f(x, args...; kwargs...)
 
+# Only for internal use. Allows dispatch over anonymous functions.
+struct MissingSmallest{T}
+    lt::T
+end
+
+"""
+    missingsmallest(f)
+
+Creates a function of two arguments `x` and `y` that tests whether `x` is less
+than `y` such that `missing` is always less than the other argument. In other 
+words, returns a modified version of the partial order function `f` such that
+`missing` is the smallest possible value, and all other non-`missing` values are
+compared according to `f`.
+
+# Examples
+```
+julia> missingsmallest((x, y) -> isless(x, 10*y))(missing, Inf)
+true
+
+julia> missingsless(-Inf, missing)
+false
+```
+"""
+missingsmallest(f) = MissingSmallest(f)
+
+
+" The standard partial order `isless` modified so that `missing` is always the
+smallest possible value."
+
+"""
+    missingsmallest(x, y)
+
+The standard partial order `isless` modified so that `missing` is always the
+smallest possible value. The expected behaviour is the following:
+- If neither argument is `missing`, the function behaves exactly as `isless`.
+- If `x` is `missing` the result will be `true` regardless of the value of `y`.
+- If `y` is `missing` the result will be `false` regardless of the value of `x`.
+
+See also [`missingsmallest`](@ref), which modifies a partial order function and
+yields a function that behaves as the expected behaviour outlined above.
+
+# Examples
+```
+julia> missingsmallest(missing, Inf)
+true
+
+julia> missingsmallest(-Inf, missing)
+false
+
+julia> missingsmallest(missing, missing)
+true
+```
+"""
+missingsmallest(x, y) = missingsmallest(isless)(x, y)
+
+(ms::MissingSmallest)(x, y) = ismissing(y) ? false : ismissing(x) ? true : ms.lt(x, y)
+
+# Some optimizations
+const _MissingSmallest = Union{MissingSmallest, typeof(missingsmallest)}
+
+@static if isdefined(Base.Sort, :MissingOptimization) && isdefined(Base.Sort, :_sort!)
+    function Base.Sort._sort!(v::AbstractVector, a::Base.Sort.MissingOptimization,
+        o::Base.Order.Lt{<:_MissingSmallest}, kw)
+        # put missing at beginning
+        Base.Sort._sort!(v, a.next, withoutmissingordering(o), kw...;hi=new_hi)
```

Comment:
This is pseudocode, it needs to actually put the missing values at the beginning.

Reply:
Or just remove the optimizations.
And if there are optimizations, there should be tests that cover those branches.

Reply:
as discussed earlier - let us start with the simplest possible definition that does not try to do optimizations.